src/main/scala/org/constellation/domain/p2p/PeerHealthCheck.scala

```diff
@@ -63,18 +73,53 @@ class PeerHealthCheck[F[_]](cluster: Cluster[F], apiClient: ClientInterpreter[F]
 
   private def checkPeer(peer: PeerClientMetadata, timeout: FiniteDuration): F[PeerHealthCheckStatus] =
     timeoutTo(
-      apiClient.metrics
-        .checkHealth()
-        .run(peer)
-        .map(_ => PeerAvailable.asInstanceOf[PeerHealthCheckStatus])
-        .handleErrorWith(_ => PeerUnresponsive.asInstanceOf[PeerHealthCheckStatus].pure[F]),
+      {
+        val check: F[PeerHealthCheckStatus] = apiClient.metrics
+          .checkHealth()
+          .run(peer)
+          .flatMap(_ => getClock)
+          .map[PeerHealthCheckStatus](clock => PeerAvailable(clock))
+          .flatTap { status =>
+            state.modify { m =>
+              val updated = m.updated(peer.id, status)
+              (updated, ())
+            }
+          }
+          .handleErrorWith(
+            _ =>
+              C.monotonic(MILLISECONDS).flatMap { clock =>
```

Comment:
How about reusing `getClock` here? (Unless you need another instance of clock)

Reply:
:+1:

Reply:
I have always a new instance of clock because it's lazy evaluated